### PR TITLE
Fuzzers: Set up a fake browser environment for FuzzCSSParser

### LINKED ISSUE #21629: Fuzzers: CSS fuzzer crashes on all inputs
`FuzzCSSParser` currently crashes on any input it is given.

Can be reproduced with:

```sh
> ./Meta/Lagom/BuildFuzzers.sh
> ./Meta/Lagom/Build/lagom-fuzzers/bin/FuzzCSSParser
```

which results in the following stack trace:

```
    0x7fc71a7d55d8 in ak_verification_failed /home/tim/repos/serenity/AK/Assertions.cpp:106:5
    0x559a98ffb120 in AK::Vector<JS::ExecutionContext*, 0ul>::at(unsigned long) /home/tim/repos/serenity/Meta/Lagom/../../AK/Vector.h:148:9
    0x559a98ffb120 in AK::Vector<JS::ExecutionContext*, 0ul>::last() /home/tim/repos/serenity/Meta/Lagom/../../AK/Vector.h:162:34
    0x559a98ffac9d in JS::VM::running_execution_context() /home/tim/repos/serenity/Meta/Lagom/../../Userland/Libraries/LibJS/Runtime/VM.h:114:87
    0x559a98ffac9d in JS::VM::current_realm() /home/tim/repos/serenity/Meta/Lagom/../../Userland/Libraries/LibJS/Runtime/VM.h:131:37
    0x559a98ffac9d in LLVMFuzzerTestOneInput /home/tim/repos/serenity/Meta/Lagom/Fuzzers/FuzzCSSParser.cpp:29:74
    0x559a98e9ea78 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (/home/tim/repos/serenity/Meta/Lagom/Build/lagom-fuzzers/bin/FuzzCSSParser+0x57a78) (BuildId: 739f6ddedcfe7f3e13ab8f19f820a4401597fe19)
    0x559a98ea0d25 in fuzzer::Fuzzer::ReadAndExecuteSeedCorpora(std::vector<fuzzer::SizedFile, std::allocator<fuzzer::SizedFile>>&) (/home/tim/repos/serenity/Meta/Lagom/Build/lagom-fuzzers/bin/FuzzCSSParser+0x59d25) (BuildId: 739f6ddedcfe7f3e13ab8f19f820a4401597fe19)
    0x559a98ea13fe in fuzzer::Fuzzer::Loop(std::vector<fuzzer::SizedFile, std::allocator<fuzzer::SizedFile>>&) (/home/tim/repos/serenity/Meta/Lagom/Build/lagom-fuzzers/bin/FuzzCSSParser+0x5a3fe) (BuildId: 739f6ddedcfe7f3e13ab8f19f820a4401597fe19)
    0x559a98e81af2 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) (/home/tim/repos/serenity/Meta/Lagom/Build/lagom-fuzzers/bin/FuzzCSSParser+0x3aaf2) (BuildId: 739f6ddedcfe7f3e13ab8f19f820a4401597fe19)
    0x559a98e6b557 in main (/home/tim/repos/serenity/Meta/Lagom/Build/lagom-fuzzers/bin/FuzzCSSParser+0x24557) (BuildId: 739f6ddedcfe7f3e13ab8f19f820a4401597fe19)
    0x7fc71a245ccf  (/usr/lib/libc.so.6+0x27ccf) (BuildId: 8bfe03f6bf9b6a6e2591babd0bbc266837d8f658)
    0x7fc71a245d89 in __libc_start_main (/usr/lib/libc.so.6+0x27d89) (BuildId: 8bfe03f6bf9b6a6e2591babd0bbc266837d8f658)
    0x559a98e6b894 in _start (/home/tim/repos/serenity/Meta/Lagom/Build/lagom-fuzzers/bin/FuzzCSSParser+0x24894) (BuildId: 739f6ddedcfe7f3e13ab8f19f820a4401597fe19)
```

### PR DESCRIPTION
**Pull request by AtkinsSJ**
WIP! Looking for feedback from whoever actually understands any of the parts involved, unlike myself. :sweat_smile: 

This is likely not super correct, but it does at least function for me now. (As far as I can tell - I don't know what FuzzFoo output is supposed to look like, but it seems happy.) The remaining issues are that I've hard-coded the fonts directory and default font query.

It's possible we could avoid needing font information entirely by not creating a Document but doing things manually instead? Not sure. This feels like way too much is being set up that we don't actually use - we just need a JS realm with the CSSOM intrinsics available.

Fixes #21629.
Well, kinda.

### CODE REVIEW

File: Meta/Lagom/Fuzzers/FuzzCSSParser.cpp [hunk 1]
```diff
@@ -16,16 +22,68 @@ struct Globals {
 
 Globals::Globals()
 {
+    // FIXME: Somehow pick a suitable font location and default query
+    Core::ResourceImplementation::install(make<Core::ResourceImplementationFile>("/usr/share/fonts"_string));
+    Gfx::FontDatabase::set_default_font_query("Liberation Sans 10 400 0");
+    Gfx::FontDatabase::the().load_all_fonts_from_uri("resource:///"sv);
```
Comment (AtkinsSJ) [inline Meta/Lagom/Fuzzers/FuzzCSSParser.cpp:28]: I'm guessing a FontPlugin would solve this problem?

Comment (ADKaster) [reply Meta/Lagom/Fuzzers/FuzzCSSParser.cpp:28]: This is pretty unfortunate that we need fonts in order to fuzz CSS :thonk:

We could play games like copying the LiberationSerif or SerenitySans font files into the build directory, and load relative to Core::Process::current_executable_path(). But we would need to make sure that the CMakeLists.txt and the BuildFuzzers.sh script and the oss-fuzz Dockerfile (in the oss-fuzz repo) end up copying them into the oss-fuzz test env properly.

Comment (ADKaster) [reply Meta/Lagom/Fuzzers/FuzzCSSParser.cpp:28]: Loading fonts from the system is also kind of spooky because it makes the tests less reproducible

**Comment by AtkinsSJ** [comment]
Not stale! Though I had forgotten about this. 😅

**Comment by AtkinsSJ** [comment]
Same again. I can't believe this has been open 2 months. :thousandyakstare:

**Comment by timschumi** [comment]
It's a bit sad that we have to do all this extra work, because some (more advanced) fuzzers actually trace inputs to get better results (and this will almost certainly throw them off a bit). Although, arguably, an inefficient fuzzer would still be a lot better than the current one.

Is there a chance that we can move any of the things before `parse_css_stylesheet` into the static initialization part? (And even that only helps the fuzzers that understand the way of libFuzzer, not sure if that includes oss-fuzz.)

Other than that, it looks OK, I guess...

**Comment by ADKaster** [comment]
Stalebot pls

**Comment by AtkinsSJ** [comment]
I keep forgetting about this. 😅 (And possibly avoiding it...)

**Pull request closed without merging**